ReactCommon/react/renderer/mapbuffer/MapBuffer.cpp

```diff
@@ -12,14 +12,21 @@ using namespace facebook::react;
 namespace facebook {
 namespace react {
 
-MapBuffer::MapBuffer(int initialSize) {
+MapBuffer::MapBuffer(uint16_t initialSize) {
   _dataSize = initialSize;
   _data = new Byte[_dataSize];
   // TODO: Should we clean up memory here?
 }
 
 void MapBuffer::makeSpace() {
   int oldDataSize = _dataSize;
+  if (_dataSize >= std::numeric_limits<uint16_t>::max() / 2) {
+    LOG(ERROR)
+        << "Error: trying to assign a value beyond the capacity of uint16_t"
+        << _dataSize * 2;
```

Comment:
this will overflow, so cast to something bigger before multiplying :) (same below)

Reply:
This is quite hilarious..